engine/utils/file_manager_utils.h

```diff
@@ -50,15 +50,84 @@ inline std::filesystem::path GetExecutableFolderContainerPath() {
 #endif
 }
 
+inline std::filesystem::path GetHomeDirectoryPath() {
+#ifdef _WIN32
+  const char* homeDir = std::getenv("USERPROFILE");
+  if (!homeDir) {
+    // Fallback if USERPROFILE is not set
+    const char* homeDrive = std::getenv("HOMEDRIVE");
+    const char* homePath = std::getenv("HOMEPATH");
+    if (homeDrive && homePath) {
+      return std::filesystem::path(homeDrive) / std::filesystem::path(homePath);
+    } else {
+      throw std::runtime_error("Cannot determine the home directory");
+    }
+  }
+#else
+  const char* homeDir = std::getenv("HOME");
+  if (!homeDir) {
+    throw std::runtime_error("Cannot determine the home directory");
+  }
+#endif
+  return std::filesystem::path(homeDir);
+}
+
+inline std::filesystem::path GetConfigurationPath() {
+  auto home_path = GetHomeDirectoryPath();
```

Comment:
Just for debugging purpose, can we add a flag when configure project to customize the configuration file name? Something like: -DCORTEX_CONFIG_FILE_NAME
If it is not defined, will use `.cortexrc` file